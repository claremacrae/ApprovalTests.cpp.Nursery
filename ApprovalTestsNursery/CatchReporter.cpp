#include "CatchReporter.h"
#include "FileUtilsAdditions.h"

#include <iostream>

// TODO Need to work out how to include this in a release, without
//      giving a compilation error for those using Google Test Framework
#include "Catch.hpp"

bool CatchReporter::report(std::string received, std::string approved) const
{
    const auto receivedText = FileUtilsAdditions::readFileReturnEmptyIfMissing(received);
    const auto approvedText = FileUtilsAdditions::readFileReturnEmptyIfMissing(approved);
    // By using CHECK instead of REQUIRE, program flow continues - Catch does not throw an exception
    std::cout << "Checking content of files:" << std::endl;
    std::cout << "received: " << received << std::endl;
    std::cout << "approved: " << approved << std::endl;
    CHECK( receivedText == approvedText );
    return true;
}
