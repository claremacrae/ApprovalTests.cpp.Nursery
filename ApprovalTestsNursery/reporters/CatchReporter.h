#ifndef APPROVALTESTS_CPP_NURSERY_CATCHREPORTER_H
#define APPROVALTESTS_CPP_NURSERY_CATCHREPORTER_H

#include "reporters/Reporter.h"

// A reporter which uses Catch CHECK statement to check the output
class CatchReporter : public Reporter
{
public:
    bool report(std::string received, std::string approved) const override;
};


#endif //APPROVALTESTS_CPP_NURSERY_CATCHREPORTER_H
