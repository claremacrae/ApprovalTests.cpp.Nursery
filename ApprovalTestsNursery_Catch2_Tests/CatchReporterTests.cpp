#include "CatchReporter.h"
#include "ApprovalTests/Approvals.h"

#include "Catch.hpp"

TEST_CASE("CatchReporter logs failure", "[!shouldfail]")
{
    CatchReporter reporter;

    try
    {
       Approvals::verify("This test is expected to fail", reporter);
    }
    catch(const ApprovalException& e)
    {
        // We intend to arrive here
        SUCCEED("CatchReporter reported a failure, as intended");
    }
}
