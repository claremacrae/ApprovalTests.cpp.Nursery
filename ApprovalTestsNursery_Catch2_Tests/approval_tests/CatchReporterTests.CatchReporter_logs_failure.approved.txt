Do not approve this - we want received and approved to differ.
